packages/plugins/typed-modal/src/Modal.js

```diff
@@ -0,0 +1,366 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Class responsible for creating a Blockly modal.
+ * @author aschmiedt@google.com (Abby Schmiedt)
+ */
+
+
+import * as Blockly from 'blockly/core';
+import { injectCss } from "./css_inject";
+
+/**
+ * Class responsible for creating a Blockly modal.
+ */
+export class Modal {
+
+  /**
+   * Constructor for creating a Blockly modal.
+   * @param {string} title The title for the modal.
+   * @param {!Blockly.WorkspaceSvg} workspace The workspace to display the modal
+   *     over.
+   */
+  constructor(title, workspace) {
+    /**
+     * The title for the modal.
+     * @type {string}
+     * @private
+     */
+    this.title_ = title;
+
+    /**
+     * The workspace to display the modal over.
+     * @type {!Blockly.WorkspaceSvg}
+     * @protected
+     */
+    this.workspace_ = workspace;
+
+    /**
+     * The list of focusable elements in the modal.
+     * @type {Array<!HTMLElement>}
+     * @private
+     */
+    this.focusableEls = [];
+
+    /**
+     * HTML container for the modal.
+     * @type {HTMLDivElement}
+     * @private
+     */
+    this.htmlDiv_ = null;
+
+    /**
+     * Array holding info needed to unbind events.
+     * Used for disposing.
+     * Ex: [[node, name, func], [node, name, func]].
+     * @type {!Array.<Array<?>>}
+     * @private
+     */
+    this.boundEvents_ = [];
+
+    /**
+     * If true close the modal when the user clicks outside the modal.
+     * Otherwise, only close when user hits the 'X' button or escape.
+     * @type {boolean}
+     */
+    this.closeOnClick = true;
+  }
+
+  /**
+   * Initialize a Blockly modal.
+   */
+  init() {
+    this.injectCss_();
+    this.render();
+    this.addEvent_(/** @type{!HTMLDivElement} */ this.htmlDiv_, 'keydown',
+        this, this.handleKeyDown_);
+  }
+
+  /**
+   * Inject the css for a Blockly modal.
+   * @protected
+   */
+  injectCss_() {
+    injectCss('blockly-modal-css', `
```

Comment:
I still think there's little reason not to use Blockly.CSS.register. We're shipping css injection logic in every plugin and that doesn't make sense.

Reply:
You should be able to add the CSS to be registered during page load, rather than during modal initialization, like the toolbox in Blockly for example: 
https://github.com/google/blockly/blob/master/core/toolbox.js#L741